radis/spectrum/spectrum.py

```diff
@@ -2137,26 +2140,46 @@ def clean_error_msg(string):
         # (useful when plotting multiple plots on same figure)
         label = kwargs.pop("label", self.get_name())
 
+
+        ax=fig.gca()
+
         # Actual plot :
         # ... note: '-k' by default with style origin for first plot
         if not plot_by_parts:
-            (line,) = plt.plot(x, y, label=label, **kwargs)
+            (line,) = ax.plot(x, y, label=label, **kwargs)
         else:
             (line,) = split_and_plot_by_parts(x, y, ax=fig.gca(), label=label, **kwargs)
             # note: split_and_plot_by_parts pops 'cutwing' & 'split_threshold' from kwargs
 
         if show_points:
-            plt.plot(x, y, "o", color="lightgrey", **kwargs)
+            ax.plot(x, y, "o", color="lightgrey", **kwargs)
 
+    
         # Labels
-        plt.ticklabel_format(useOffset=False, axis="x")
-        plt.xlabel(xlabel)
-        plt.ylabel(ylabel)
-        plt.yscale(yscale)
+        ax.ticklabel_format(useOffset=False, axis="x")
+        ax.set_xlabel(xlabel)
+        ax.set_ylabel(ylabel)
+        ax.set_yscale(yscale)
         if "label" in kwargs:
-            plt.legend()
+            ax.legend()
         fix_style()
 
+        def waveLength2waveNumber(x):
+            return 1e4/x
```

Comment:
There are cm2nm and nm2cm functions in RADIS (.phys.convert maybe?), can you use them? 

Btw you might find a problem : wavelength in vacuum are slightly different from wavelength in air. Assume wavelength in air if wunit is "nm" or "nm_air", assume wavelength in vacuum if "nm_vac"

Reply:
sure, i'll update the code using the functions defined in the .physics.convert and also take care to of wavelength in air and vaccum